specs/phase0/p2p-interface.md

```diff
@@ -745,10 +747,27 @@ The request MUST be encoded as an SSZ-container.
 The response MUST consist of zero or more `response_chunk`.
 Each _successful_ `response_chunk` MUST contain a single `SignedBeaconBlock` payload.
 
-Clients MUST keep a record of signed blocks seen since the start of the weak subjectivity period
-and MUST support serving requests of blocks up to their own `head_block_root`.
+Clients MUST keep a record of signed blocks seen on the epoch range
+`[max(GENESIS_EPOCH, current_epoch - MIN_EPOCHS_FOR_BLOCK_REQUESTS), current_epoch]`
+where `current_epoch` is defined by the current wall-clock time,
+and clients MUST support serving requests of blocks on this range.
+
+Peers are *repeatedly* unable to reply to Block requests within the
+`MIN_EPOCHS_FOR_BLOCK_REQUESTS` epoch range MAY get descored or disconnected at any time.
+
+*Note*: The above requirement implies that nodes that start from a recent weak subjectivity checkpoint
+MUST backfill the local block database to at least epoch `current_epoch - MIN_EPOCHS_FOR_BLOCK_REQUESTS`
+to be fully compliant with `BlocksByRange` requests. To safely perform such a
+backfill of blocks to the recent state, the node MUST validate both (1) the
+proposer signatures and (2) that the blocks form a valid chain up to the most
+recent block referenced in the weak subjectivity state.
+
+*Note*: Although clients that bootstrap from a weak subjectivity checkpoint can begin
+participating in the networking immediately, other peers that are actively block syncing MAY
```

Comment:
```suggestion
participating in the networking immediately, other peers MAY
```

This section is somewhat contradictory - "participating in the network" entails serving blocks - also, limiting the "actively block syncing" removes the ability to detect freeloaders outside of syncing which them the opening they need, essentially. If we're going to put a `MUST` backfill, then we must also _at least_ allow disconnecting clients at any time if they do not fulfill this requirement.

There is an interesting question here what to do with read-only clients that light-sync to head but don't otherwise produce traffic - ie serving blocks and producing attestations/blocks go hand in hand because production increases network load and is therefor paid for with rewards, but "observers" do not add to traffic really.